request-server_test.go

```diff
@@ -345,6 +345,40 @@ func TestRequestFstat(t *testing.T) {
 	checkRequestServerAllocator(t, p)
 }
 
+func TestRequestFsetstat(t *testing.T) {
+	p := clientRequestServerPair(t)
+	defer p.Close()
+	_, err := putTestFile(p.cli, "/foo", "hello")
+	assert.Nil(t, err)
+	fp, err := p.cli.OpenFile("/foo", os.O_WRONLY)
+	assert.Nil(t, err)
+	err = fp.Truncate(2)
+	if assert.NoError(t, err) {
```

Comment:
Why do we have different behavior on whether there is an error or not?

We should just `require.NoError` and/or `if err != nil { t.Fatal(…) }` any time a precondition fails. There’s no point in running the rest of any of the test if these things fail.